### PR TITLE
Master

### PR DESCRIPTION
**Pull request by JHCD**
Small typo...

### CODE REVIEW

**Comment by chaorace** [comment]
The commit history on this PR isn't correct. Try checking out CQUI master to a new branch, then "git cherry-pick"ing your change over to it

**Comment by JHCD** [comment]
I doesn't understand this problem...
After deleting the fork I could do a new PR...

**Comment by chaorace** [comment]
Errr, I want to help you understand, but it's a little long winded, so please excuse me for writing a text wall here:

## Github =/= git
Github is a community for collaborating on projects that are managed using the "git" version control system. Github is a website owned by a corporation, git is an open source tool that's a core part of the Linux ecosystem (by core, I mean freakin' Linus Torvalds himself made it!).

So, let's talk a little bit about how git works first, then about how Github allows us to extend git and program collaboratively:

## git
At its core, git really only does two things: tracking changes and managing multiple versions of the same software. If you're creating a new project that you want to track, you would create a new "repo", short for repository, that holds your changes for you. These two goals of git: tracking changes and managing multiple versions, manifest as things called "commits" and "branches", respectively:
  * Commit: A snapshot of changes with a title and description
  * Branch: A single version of your work that can live in the same repository as other branches. Commits live inside of branches, just like branches live inside of a repo

So, in short, Repos hold Branches, which hold Commits. Every repo needs at least 1 branch, and every branch needs at least 1 commit. But why go though all this trouble when it's easier to just delete the old stuff and upload my new version?

Reason 1) Managing multiple versions: This is useful because it means I can have a branch for Windows _and_ a branch for Linux/OSX. Since my Branches live in the same Repo, I can copy my commits back and forth between branches automatically so that one version never falls behind the other without creating extra grunt work for me. That means I don't have to do work twice and instead git does most of the work for me

Reason 2) Keeping changes modular and documented. It's very easy to forget who did what, when, and why. It's even easier to forget to write down your changes as you make them! By organizing changes into commits, it becomes much easier to credit people properly... and occasionally berate them when they do something patently wrong.

Reason 3) Undoing mistakes. Let's say I commit [a cool new thing](ad0921a07d43fc7f649288f773fe520221c52ea9), but that thing has unintended consequences. With just one command, [I can undo that commit](66297ff30dc94f8d22b8b1cc3a8ab1bb661c5dc0), with no potential for me to accidentally delete something else too and mess things up even more.

Reason 4) Collaborating with other people. Let's say two people are both programming the same project. when they show up at work the next day with usb sticks, they compare and realize they worked on the same file last night and that they're going to have to spend the next half-hour sorting it out by hand! git, by contrast, doesn't care about files, it cares about commits. As long as they specifically didn't edit the same lines, git can apply both commits automatically without issue. If they did edit the same lines, ~~they're screwed~~ they merely need to resolve _just_ the conflicted part of the file instead of all of it

Ultimately, git is an excellent, yet complex tool that makes working with others on a single project much easier, as long as you can trust everyone with write access.

## Github

Github is a community about git. Github will host your git Repos for you and provides a bunch of accessories that make collaborating over the internet easier, like:
 * A ticket system for documenting issues with built-in git integration (this means you can reference a ticket number from git and it links to the ticket automatically, or reference a commit hash number from a ticket and it links automatically to the commit while also shortening it so it's not so damn long!)
* A wiki, for sharing knowledge and laying out the biz'
* A pretty web UI for looking at projects without having to download them first

Buuuut most importantly, Github brings us "Forks". Remember how I said you can copy changes easily between branches easily, so long as they're in the same repo? It's very useful, but the critical flaw here is that people can only collaborate with you if they too are able to write to the same repo. I can't just trust internet strangers with my repo, they could deface everything then delete the history! This is where Forks come in:

A Fork is like another layer on the git cake. So before we had Repo -> Branch -> Commit. Now we have Github -> Repo -> Branch -> Commit. So long as two repos are both on Github, they can send commits to and from each other! (With the stipulation that one of them must have been originally "Forked" from the other (for example, this CQUI repo was created by "cloning" the QUI repo, that makes CQUI a Fork of QUI. Your repo, that you cloned from CQUI to create, is also a Fork) The process for doing this called a "Pull Request", or PR for short. When you submit a PR, you are essentially creating a ticket asking the owner of the other Repo to copy (or, in git terms, "pull") commits from your repo. Anyone can create a PR, but only the owners of the repo receiving the request can say "alright, let's pull commit over". This process eliminates the trust issue of just letting anyone change your code, because really what they're doing is changing their own copy, then asking that you review the new changes.

## Pulling it all together
Okay, so what happened here? Earlier, you cloned CQUI, made your changes and pushed them back up to Github, then submitted a PR, which I approved, pulling your commit. Coolio, process works as intended... except then this time it didn't work. Why?

Well, earlier your submitted PR #338 had a style issue, so you submitted a second commit to correct this. That's fine, but it meant that if I merged your PR, I'd also be pulling over a unnecessary style commit (dbfa0c15303ce14f2c1be97a6de6a17bf8be6c4b) which would only serve to make the CQUI commit history harder to read, despite it serving a useful purpose in your clone. To resolve this issue without asking that you undo history, I used the "Squash and Merge" button, which is a handy tool for consolidating all of a PR's changes into a single commit. In short: the commit that was added to CQUI is different from the two that you asked to have us pull

Okay, so the main CQUI repo and your fork now have different histories. That's fine because there's a good reason for the changes to be 1 commit in the CQUI repo and 2 in your fork. However, since they now diverge in terms of history, if you keep submitting PRs by building upon that same diverged history, you are now sending your diverged history ALONG with your new changes, because a PR is a request to pull from your repo, it doesn't say "here's a new piece of code", it says "here's my repo, wanna copy the new commits from it?". That's why your history shows those changes you didn't actually mean to ask that we copy a second time

Alright, that's the problem, what's the solution? Branches! Let's say you have a fresh clone, it's nice and new and totally the same as the original. Instead of just writing your changes on top of it, you would instead create a new branch and make your changes there. Then, the next time you want to make another PR, you just need to create another branch from that untouched original branch and make your changes on that fresh slate instead.

"Okay, but what about if your branch changes?", you may ask. Simple, you add the main CQUI Repo as a "remote" (remote repository, like CQUI or your Fork, as opposed to the copy of your Fork which lives on your computer) and "pull" (download new changes) from there before you're about to create a new branch to write your new features on. You need to do this to keep your Fork in sync with the main CQUI Repo, because, just like you can't freely put stuff on my Repo, I can't put stuff on your Repo. You need to explicitly tell git that you want the new stuff.

# tl;dr
Here's how your workflow looks: 
  * Commit your changes to your fork
  * Submit PR

Here's how your workflow should look:
  * Pull changes from CQUI master to the main, untouched branch of your fork to be sure you're making changes on top of the latest CQUI version
  * Create a new branch from that newly updated and untouched main branch
  * Commit your changes to your newly created branch
  * Submit PR

Why should you do this? It avoids these bad history situations. It enables you to make multiple PRs at once. It will help you be more organized. It's just how things work and I'm sorry it's not easier 😭

**Comment by JHCD** [comment]
With the Linux client I could sync the changes in parent Repo to my Repo.
But it looks like this doesn't work similar with the windows client, or I haven't found it out...
I used "Update from chaorace/master" before I make my spelling change.

I will try branches now ;)
